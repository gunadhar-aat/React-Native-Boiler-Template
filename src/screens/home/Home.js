import React, { useEffect } from 'react';
import { View, Text, Pressable, StatusBar } from 'react-native';
import { useTranslation } from 'react-i18next';
import Icon from '../../components/Icon';
import { useDispatch, useSelector } from 'react-redux';
import { requestUser } from '../../store/actions/auth.actions';
import { getUserData } from '../../store/selectors/auth.selector';

const Home = ({ navigation }) => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  //user call
  useEffect(() => {
    dispatch(requestUser());
  }, []);
  const userData = useSelector(getUserData);
  // console.log(userData);
  return (
    <View>
      <StatusBar barStyle="dark-content" />
      <Text style={{ fontFamily: t('Font2'), fontSize: 30, color: '#0000FF' }}>Home Screen </Text>
      <Pressable onPress={() => navigation.navigate('Help')}>
        <Text>go to help</Text>
        <Icon pack="FontAwesome5" name={'home'} size={30} style={{ color: '#0000FF' }} />
      </Pressable>
    </View>
  );
};

export default Home;
