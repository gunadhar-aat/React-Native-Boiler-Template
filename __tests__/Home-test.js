import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/screens/Home/Home';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

test('Home SnapShot', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
