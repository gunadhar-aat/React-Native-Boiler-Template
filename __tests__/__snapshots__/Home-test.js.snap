// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home SnapShot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#0000FF",
        "fontFamily": "Font2",
        "fontSize": 30,
      }
    }
  >
    Home Screen 
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      go to help
    </Text>
  </View>
</View>
`;
